### ISSUE TITLE
How to cancel with Promise client?

### ISSUE THREAD

**Issue by emil14** [grpc/grpc-web#946]
With callback client I can do

```ts
const call = client.method(request, meta, cb)
```

While Promise client doesn't give that `call` but I need it to `call.cancel()` in the Timeout

---

BTW another question - Am I have to do that? I want to cancel my call if it exceeded the timeout, but in my `meta` I have `deadline`. Will grpc-web cancel my call for me or I have to do it manually? If it does, will promice reject with timeout error?

Thank you!

**Comment by sampajano** [comment]
Hi! Thanks for the interests here!

As mentioned in https://github.com/grpc/grpc-web/pull/1240#issuecomment-1260136406, i will try to experiment with this feature internally first soon, and upstream it to GH if it works. Thanks!

**Comment by maja42** [comment]
Any updates on this?

**Comment by sampajano** [comment]
@maja42 Hi thanks for checking!

I intended to do this but was not able to prioritize it yet..

i'll again bump this in my list and i hope to get to it in the upcoming weeks!

**Comment by kanekv** [comment]
@sampajano it's been a year since the initial PR requesting/adding it. Is there anything needed to help start moving with this thing?

**Comment by sampajano** [comment]
@kanekv You're right! I'll really try to work on this soon! Apologize for the delays!
